pkg/mutation/assignmetadata.go

```diff
@@ -0,0 +1,71 @@
+package mutation
+
+import (
+	"fmt"
+
+	"github.com/open-policy-agent/gatekeeper/pkg/mutation/path/parser"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// MetadataMutator is a mutator wrapping the AssignMetadata type.
+type AssignMetadataMutator struct {
+	//assignMetadata *mutationsv1.AssignMetadata
+	Path  parser.Path
+	Value string
+}
+
+func (m AssignMetadataMutator) Matches(obj metav1.Object, ns *corev1.Namespace) bool {
```

Comment:
Let's put a TODO here to add in the match logic

Reply:
Done
Note that this is implemented in #932